common/cuda_hip/matrix/batch_ell_kernels.hpp.inc

```diff
@@ -235,3 +235,106 @@ __global__ __launch_bounds__(default_block_size) void fill_in_dense(
         }
     }
 }
+
+template <typename IndexType>
+__global__ __launch_bounds__(default_block_size) void check_diagonal_entries(
+    const IndexType num_min_rows_cols, const size_type row_stride,
+    const size_type max_nnz_per_row,
+    const IndexType* const __restrict__ col_idxs,
+    bool* const __restrict__ has_all_diags)
+{
+    if (threadIdx.x == 0) {
+        *has_all_diags = true;
+    }
+    __syncthreads();
+    constexpr int warp_size = config::warp_size;
+    auto tile_grp =
+        group::tiled_partition<warp_size>(group::this_thread_block());
+    const auto row = thread::get_subwarp_id_flat<warp_size, IndexType>();
+    if (row == 0 && num_min_rows_cols > 0) {
```

Comment:
Maybe I missed another way to do it, but row 0 is a special case because the default column index is also zero. Since we are looking for diagonal entries, that needs to be handled separately.